qiskit/ml/algorithms/quantum_kernel.py

```diff
@@ -0,0 +1,272 @@
+
+        Raises:
+            ValueError:
+                - x_vec and/or y_vec are not two dimensional arrays
+                - x_vec and/or y_vec have incompatible dimension with feature map
+        """
+        if x_vec.ndim != 2:
+            raise ValueError("x_vec must be a 2D array")
+
+        if y_vec is not None and y_vec.ndim != 2:
+            raise ValueError("y_vec must be a 2D array")
+
+        if x_vec.shape[1] != self._feature_map.num_qubits:
+            raise ValueError("x_vec and class feature map incompatible dimensions.\n" +
+                             "x_vec has %s dimensions, but feature map has %s." %
+                             (x_vec.shape[1], self._feature_map.num_qubits))
+
+        if y_vec is not None and y_vec.shape[1] != self._feature_map.num_qubits:
+            raise ValueError("y_vec and class feature map incompatible dimensions.\n" +
+                             "y_vec has %s dimensions, but feature map has %s." %
+                             (y_vec.shape[1], self._feature_map.num_qubits))
+
+        # determine if calculating self inner product
+        is_symmetric = False
+        if y_vec is None:
+            y_vec = x_vec
+        if np.array_equal(x_vec, y_vec):
+            is_symmetric = True
+
+        # initialise kernel matrix
+        kernel = np.zeros((x_vec.shape[0], y_vec.shape[0]))
+
+        # set diagonal to 1 if symmetric
+        if is_symmetric:
+            np.fill_diagonal(kernel, 1)
+
+        # get indices to calculate
+        if is_symmetric:
+            mus, nus = np.triu_indices(x_vec.shape[0], k=1)  # remove diagonal
+        else:
+            mus, nus = np.indices((x_vec.shape[0], y_vec.shape[0]))
+            mus = np.asarray(mus.flat)
+            nus = np.asarray(nus.flat)
+
+        is_statevector_sim = self._quantum_instance.is_statevector
+
+        # calculate kernel
+        if is_statevector_sim:  # using state vector simulator
+            if is_symmetric:
+                to_be_computed_data = x_vec
+            else:  # not symmetric
+                to_be_computed_data = np.concatenate((x_vec, y_vec))
+
+            feature_map_params = ParameterVector('par_x', self._feature_map.num_parameters)
+            parameterized_circuit = self._construct_circuit(
+                feature_map_params, feature_map_params,
+                is_statevector_sim=is_statevector_sim)
+            parameterized_circuit = self._quantum_instance.transpile(parameterized_circuit)[0]
+            circuits = [parameterized_circuit.assign_parameters({feature_map_params: x})
+                        for x in to_be_computed_data]
+
+            results = self._quantum_instance.execute(circuits)
+
+            offset = 0 if is_symmetric else len(x_vec)
+            matrix_elements = [self._compute_overlap(idx, results, is_statevector_sim)
+                               for idx in list(zip(mus, nus + offset))]
+
+            # using parallel_map causes an error
```

Comment:
What is the error? Are you using a Mac and getting BrokenProcessPool? I am not sure how much benefit having this done in a parallel truly is. If it ends up not going to be in the code and doesn't work I think this should be eliminated rather than exist here commented out

Reply:
Removed in 184acd5